[tahnya] iP

tahnya opened the pull request:
### Signal Bot

Signal is a chatbot designed to help you keep track of your tasks! It's

- text based
- easy to learn
- _super_ fast to use!

> “Complaining will not change things in your life; the only action will. Make a list of all you need to do to change what doesn’t work in your life and, little by little, begin making those changes.” — Susan Jeffers ([source](https://medium.com/@oaklinejournal/25-quotes-from-successful-people-on-the-power-of-writing-to-do-lists-8a0b3729a34c))

Change your life today! Click [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ) for a demo! 😜

Let Signal help you manage your tasks:

1. It's quick
2. easy
3. and **FREE!**

Some features:

- [ ] Manage tasks
- [ ] Manage deadlines (coming soon)
- [ ] View events (coming soon)
- [ ] Reminders (coming soon)

Here's some commands:

1. Create tasks:
`todo <>`
`deadline <> /by <>`
`event <> /from <> /to <>`

2. List the tasks added:
`list`

3. Mark a task
`mark <>`
`unmark <>`

4. Delete a task
`delete <>`

5. Find a task
`find <>`

Here's an excerpt of the main method:
```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

tahnya committed:
Level 0. Rename, Greet, Exit

tahnya committed:
Level 1. Echo

tahnya committed:
Level 2. Add, List

tahnya committed:
Level 3. Mark as Done

tahnya committed:
Level 4. ToDos, Events, Deadlines

tahnya committed:
Level 5. Handle Errors

tahnya committed:
Level 6. Delete

tahnya committed:
Edit delete

tahnya committed:
Edit help

tahnya committed:
Fixed delete

tahnya committed:
Modify taskList from list to ArrayList

tahnya committed:
Level 7. Save

tahnya committed:
Level 8. Dates and Times

tahnya committed:
Update dates

tahnya committed:
Merge remote-tracking branch 'origin/add-gradle-support'

tankh99 commented on src/main/java/Duke.java:
In general, it's not advisable to put all your code in one file for a majority of reasons, mainly due to extensibility and maintainability

tankh99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+import com.sun.source.util.TaskListener;
+
+import java.io.*;
```
Avoid wildcard imports

tankh99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+        String formattedTime = time.format(formatter);
+        return formattedTime;
+    }
+    public static class Task {
```
Creating a nested class in Duke couples the logic together, and you're essentially saying that Task cannot exist without Duke

tankh99 replied:
Also, why did you declare this as static?

tankh99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+        return formattedTime;
+    }
+    public static class Task {
+        protected String description;
+        protected boolean isDone;
```
Why not use private?

tankh99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+        }
+
+        public String toString() {
+            return " | " + getStatusIcon() + " | " + this.description;
```
Shouldn't this be an overridden method?

tankh99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+        public boolean checkDone() {
+            return this.isDone;
+        }
+        public void markDone() {
+            this.isDone = true;
+        }
+
+        public void markUnDone() {
+            this.isDone = false;
+        }
```
You can merge these 2 functions into a single function, just by taking a boolean parameter

tankh99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+            super(description, isDone);
+        }
+
+        @Override
+        public String toString() {
+            return "T" + super.toString();
+        }
```
I like the usage of superclass methods here

tankh99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+    }
+
+    public static class Deadline extends Task {
+//        protected String by;
```
nit: unnecessary comment

tankh99 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+                signalSays(commandUnmark(itemIndex));
+            } else if (inputArray.length == 2 && (isPermutationMatch(inputArray[0], "mark") || isPermutationMatch(inputArray[0], "unmark"))) {
+                // check typo of "mark" or "unmark"
+                if (checkCommandTypo(inputArray[0], "mark")) { // command mark typo
+                    int itemIndex = Integer.parseInt(inputArray[1]) - 1;
+                    signalSays(commandMark(itemIndex));
+                } else if (checkCommandTypo(inputArray[0], "unmark")) { // command unmark typo
+                    int itemIndex = Integer.parseInt(inputArray[1]) - 1;
+                    signalSays(commandUnmark(itemIndex));
```
I don't fully understand the purpose of `checkCommandTypo`. Isn't it always going to trigger when you type "unmark" since it checks for "mark" first?

tankh99 reviewed (commented):
I think it's a good idea in general to separate logic into different files. This makes it easier for other developers to read and understand your code. Also found a few nits here and there

yap-zong-xin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+        index += 1;
+        signalSays("Added: " + input);
+    }
+
```
Not sure if you are done with the JavaDocs, but it would be great to include them for all the methods. If not, your other JavaDocs look great to me!

yap-zong-xin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+import java.time.temporal.TemporalAdjusters;
+import java.util.ArrayList;
```
Just a small note: Good to separate the different packages (e.g. time, util, io) with an empty line :)

yap-zong-xin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+
+    }
+
+    private static final String LOGO = " _______  __                       __ \n"
+            + "|     __||__|.-----..-----..---.-.|  |\n"
+            + "|__     ||  ||  _  ||     ||  _  ||  |\n"
+            + "|_______||__||___  ||__|__||___._||__|\n"
+            + "             |_____|                  \n";
```
I like that you created a new logo! :D

yap-zong-xin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,585 @@
+//                    commandListDate(inputArray[1]);
+//                }
+//            }
+            else if (isPermutationMatch(userInput, "list")) {
+                // check typo of command list
+                if (checkCommandTypo(userInput, "list")) {
+                    commandList();
+                } else {
+                    signalSays("Do you want to add " + userInput + "? (y/n)");
+                    String addCommandCheck = scanner.nextLine();
+                    if(addCommandCheck.equals("n")) {
+                        signalSays("What else can I help you with?");
+                    } else if(addCommandCheck.equals("y")) {
+                        taskAdded(userInput);
+                    }
+                }
```
It would be better if you have avoided 'deep nesting' within your methods

yap-zong-xin reviewed (commented):
LGTM! Just need to clean up your codes that's all, keep up the good work and jiayou!🥇

tahnya committed:
Moved Task classes to separate files

tahnya committed:
Split up code into separate files

tahnya committed:
Added JavaDocs

tahnya committed:
Add comments

tahnya committed:
Add find

tahnya committed:
Merge tag 'Level-9'

tahnya committed:
Fix find

tahnya committed:
Remove commented out code

tahnya committed:
Set up JavaFX

tahnya committed:
Update TimeManager

tahnya committed:
Create helloWorld GUI

tahnya committed:
Update GUI to echo the user

tahnya committed:
Add Dialog boxes and avatar images

tahnya committed:
Update dialog box

tahnya committed:
Update dialog box to flip the view

tahnya committed:
Add dialog box styling

tahnya committed:
Apply coding standard

tahnya committed:
Add JavaDoc to DialogBox

tahnya committed:
Update styling and add intro message

tahnya committed:
Use FXML

tahnya committed:
Update GUI style

tahnya committed:
Update dialogbox

tahnya committed:
Update style and command calls

tahnya committed:
Update user commands to show on GUI

tahnya committed:
Generate JAR file

tahnya committed:
Add assertions

Added assertions to places in the code that are assumed to always be true.

In Parser, it is assumed that the reply returned always has a message from Signal.

In Ui, it is assumed that the task to add in taskAdded() is not null.

tahnya committed:
Update signalSays

Reduce to one instance of signalSays in the read() method in Parser to show the reply,
instead of calling the method in every command function call.

Removed commented out code

tahnya committed:
Update find command

commandFind throws DukeException when user inputs 'find' with no keyword

tahnya committed:
Merge pull request #2 from tahnya/branch-A-Assertions

Add assertions

tahnya committed:
Merge branch 'master' into branch-A-CodeQuality

tahnya committed:
Merge pull request #3 from tahnya/branch-A-CodeQuality

Edit code quality

tahnya committed:
Update textbox style

Make width of text box responsive to width of text up to
a maximum width of 260px

Set minimum height of text to negative infinity to prevent
cut off of text

tahnya committed:
Update style

Changed max width of textbox

Prints intro message

tahnya committed:
Add prioritise, unprioritise, prioritylist and notdonelist

Add isPriority variable to Task.

Edit parser to parse new commands above.

Add methods to Ui to set priority of tasks, return the list of
prioritised tasks and return list of tasks not done yet.

tahnya committed:
Update help function

tahnya committed:
Update README.md

tahnya committed:
Add screenshot

Picture of the chatbot

tahnya committed:
Merge branch 'master' of https://github.com/tahnya/ip

tahnya committed:
Edit README.md

tahnya committed:
Update README.md

tahnya committed:
Change displayPicture size

tahnya committed:
Make window resizeable

tahnya committed:
Refactor Duke to Signal

Add function to close window when command 'bye' is input

tahnya committed:
Add message on close request

tahnya committed:
Fix markDone bug

tahnya committed:
Add Ui.png and change displaypicture size

tahnya committed:
Update Ui.png

tahnya committed:
Update Ui.png

tahnya committed:
Update Duke to Signal

tahnya committed:
Add MainWindow responsiveness

tahnya committed:
Remove dead code

tahnya committed:
Fix event bug

tahnya committed:
Fix list bug

tahnya committed:
Add test input output

tahnya committed:
Add JUnit testing for creating tasks and finding tasks

tahnya committed:
Account for empty task description

tahnya committed:
Check for wrong date format

tahnya committed:
Add constructors without arguments

tahnya committed:
Add JavaDocs

tahnya committed:
Update return string

tahnya committed:
Add UG

tahnya committed:
Update README with UG link

tahnya committed:
Update date format

tahnya committed:
Update ReadME

tahnya committed:
Add JavaDocs

tahnya committed:
Fix bug for typo feature

tahnya committed:
Fix jar bug
